tests/pytest/test_np_views.py

```diff
@@ -25,48 +28,86 @@ def check_leak(func, tol=1.e6):
         raise MemoryError("Function did not correctly clean up")
     else:
         print("Function %s: PASSED" % func.__name__)
+        return True
+
 
 def build_mat():
     mat = psi4.core.Matrix(mat_size, mat_size)
     return mat
 
+
 def build_view_mat():
     mat = psi4.core.Matrix(mat_size, mat_size)
     view = mat.np
     return mat, view
 
+
 def build_viewh_mat():
     mat = psi4.core.Matrix(mat_size, mat_size)
     view = mat.np
     return mat, view
 
+
 def build_view_set_mat():
     mat = psi4.core.Matrix(mat_size, mat_size)
     view = mat.np
     view[:] = 5
     return mat, view
 
+
 def build_arr_mat():
     mat = psi4.core.Matrix(mat_size, mat_size)
     view = np.asarray(mat)
     return mat, view
 
+
 def build_copy_mat():
     mat = psi4.core.Matrix(mat_size, mat_size)
     view = np.array(mat)
     return mat, view
 
-if __name__ == "__main__":
-    # Run the checks
+
+@using_memory_profiler
+def test_build_mat():
+    assert(check_leak(build_mat))
+
+
+@using_memory_profiler
+def test_build_view_mat():
+    assert(check_leak(build_view_mat))
+
+
+@using_memory_profiler
+def test_build_viewh_mat():
+    assert(check_leak(build_viewh_mat))
+
+
+@using_memory_profiler
+def test_build_view_set_mat():
+    assert(check_leak(build_view_set_mat))
+
+
+@using_memory_profiler
+def test_build_arr_mat():
+    assert(check_leak(build_arr_mat))
+
+
+@using_memory_profiler
+def test_build_copy_mat():
+    assert(check_leak(build_copy_mat))
+
+
+@using_memory_profiler
+def test_totals():
     start = snapshot_memory()
```

Comment:
Didn't know about `memory_profiler` before.  Neat!

Reply:
Me, neither, nor the `mpmath` for multiprecision math that DGAS used for gau2grid.

Reply:
The wonder of Python, I have yet to be left wanting for a library that did not involve quantum mechanics in some way.